RubberduckTests/Symbols/DeclarationFinderTests.cs

```diff
@@ -1303,6 +1304,667 @@ End Sub
+            var vbe = new MockVbeBuilder()
+                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+                .AddProjectToVbeBuilder()
+                .Build();
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var declarations = state.DeclarationFinder.AllDeclarations.ToList();
+                var implementing = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertyLet);
+
+                Assert.IsFalse((implementing as ModuleBodyElementDeclaration)?.IsInterfaceImplementation);
+            }
+        }
+
+        [Test]
+        [Category("Resolver")]
+        [Category("Interfaces")]
+        public void ImplementsInterfaceMemberVariantMatchesLetAndSet()
+        {
+            var intrface =
+                @"Option Explicit
+
+Public Foo As Variant";
+
+            var implementation =
+                @"Option Explicit
+
+Implements TestInterface
+
+Public Property Get TestInterface_Foo() As Variant
+End Property
+
+Public Property Let TestInterface_Foo(rhs As Variant)
+End Property
+
+Public Property Set TestInterface_Foo(rhs As Variant)
+End Property
+";
+            var vbe = new MockVbeBuilder()
+                .ProjectBuilder("UnderTest", ProjectProtection.Unprotected)
+                .AddComponent("TestInterface", ComponentType.ClassModule, intrface, new Selection(1, 1))
+                .AddComponent("TestImplementation", ComponentType.ClassModule, implementation, new Selection(1, 1))
+                .AddProjectToVbeBuilder()
+                .Build();
+
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var declarations = state.DeclarationFinder.AllDeclarations.ToList();
+                var expected = declarations.Single(decl => decl.IdentifierName.Equals("Foo") && decl.DeclarationType == DeclarationType.Variable);
+                var setter = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertySet);
+                var letter = declarations.Single(decl => decl.IdentifierName.Equals("TestInterface_Foo") && decl.DeclarationType == DeclarationType.PropertyLet);
+
+                var actualSetter = (setter as ModuleBodyElementDeclaration)?.InterfaceMemberImplemented;
+                var actualLetter = (letter as ModuleBodyElementDeclaration)?.InterfaceMemberImplemented;
+
+                Assert.AreEqual(expected, actualSetter, "Expected {0}, resolved to {1}", expected, actualSetter);
+                Assert.AreEqual(expected, actualLetter, "Expected {0}, resolved to {1}", expected, actualLetter);
+            }
+        }
+
```

Comment:
Don't you think we need an unit test to verify that we can still match the property even if the name is different? All unit tests use `rhs` or `NewValue`; will it still find if the definition uses `NewValue` while the implementation uses `rhs` instead?

Reply:
Yep, I didn't think to add that because it doesn't even look at the parameters (if there's a type or signature mismatch that's a compile error IIR).  I can add it a test though to make sure that the naming doesn't matter.